src/components/ProductsDashboard/AddProduct/ProductForm/ProductForm.js

```diff
@@ -34,39 +39,36 @@ const renderField = ({
   <React.Fragment>
     <input {...input} type={type} />
     <div>
-      {
-        touched
-        && error
-        && <span>{error}</span>
-      }
+      {touched && error && <span>{error}</span>}
     </div>
   </React.Fragment>
 );
 
+const renderAuction = (fields, error) => (auction, index) => (
```

Comment:
Very nice usage of nested functions. 👍 
We can simplify this a bit (from the calling perspective) because `fields` are passed as a third argument by the `map` function.
Please check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#Syntax
so
`const renderAuction = (error) => (auction, index, fields) => (`
That way when we call we can just
`{fields.map(renderAuction(error))}`